classify_videos.py

```diff
@@ -86,25 +86,34 @@
 print(f"Using device: {device}")
 
 checkpoint = "HuggingFaceM4/idefics-9b-instruct"
-model = IdeficsForVisionText2Text.from_pretrained(checkpoint, torch_dtype=torch.float16).to(device)
-processor = AutoProcessor.from_pretrained(checkpoint)
+cache_dir = ".cache"
+model = IdeficsForVisionText2Text.from_pretrained(checkpoint, torch_dtype=torch.bfloat16, cache_dir=cache_dir).to(
+    device
+)
+processor = AutoProcessor.from_pretrained(checkpoint, cache_dir=cache_dir)
 
+
+prompts = []
 for video in selected_frames.keys():
-    print(f"analyzing video {video}...")
-
-image1 = Image.open("frames/frame_0000.jpg")
-image2 = Image.open("frames/frame_0001.jpg")
-prompts = [  #
-    SYSTEM_PROMPT
-    + [
-        "\nUser:",
-        image1,
-        image2,
-        "These are a few screenshots from a tiktok video.  As these are just a few screenshots, please do not assume that the video is, for example, a collage.  Please tell me in a few words each what the topic and style of the video might be.",
-        "<end_of_utterance>",
-        "\nAssistant:",
-    ],
-]
+    PROMPT = """
+    These are a few frames from a tiktok video. Write a one paragraph description of the video.
```

Comment:
Does it not work better if you explicitly tell it how many frames you're providing?

Reply:
Let's try

Reply:
Seems to improve